force-app/main/default/classes/Grid.cls

```diff
@@ -0,0 +1,124 @@
+/**
+* @description tracks the position of a user in a 5x5 grid as x and y coordinates. 
+* The coordinates are index origin 0. The default position is the top left,
+* but the user can specify the position in the constructor.
+*/
+public with sharing class Grid 
+{
+    // holds the x coordinate within the 5x5 grid
+    public Integer x
+    {
+        get { return x; }
+        set 
+        { 
+            if (value < 0) 
+            {
+                x = 0;
+            } 
+            else if (value > 4)
```

Comment:
This would be better as a constant rather than a magic number.

Reply:
created a constant to hold the grid size